### PR TITLE
Add libc++ data formatters for std::optional.

### PR DESCRIPTION
**Pull request by shafik**
Patch by Shafik Yaghmour!

This reapplies an earlier version after addressing some post-commit feedback.

Differential Revision: https://reviews.llvm.org/D49271

git-svn-id: https://llvm.org/svn/llvm-project/lldb/trunk@339828 91177308-0d34-0410-b5e6-96231b3b80d8

### CODE REVIEW

**Comment by shafik** [comment]
@swift-ci test

**Comment by shafik** [comment]
@swift-ci test

File: source/Plugins/Language/CPlusPlus/CPlusPlusLanguage.cpp [hunk 1]
```diff
@@ -251,14 +251,14 @@ std::string CPlusPlusLanguage::MethodName::GetScopeQualifiedName() {
 bool CPlusPlusLanguage::IsCPPMangledName(const char *name) {
   // FIXME!! we should really run through all the known C++ Language plugins
   // and ask each one if this is a C++ mangled name
-  
+
```
Comment (dcci) [inline source/Plugins/Language/CPlusPlus/CPlusPlusLanguage.cpp:254]: This looks like a bunch of whitespace fixing that's unrelated. Can you revert?

File: source/Plugins/Language/CPlusPlus/LibCxx.cpp [hunk 2]
```diff
@@ -120,9 +142,9 @@ bool lldb_private::formatters::LibCxxMapIteratorSyntheticFrontEnd::Update() {
 
   if (!valobj_sp)
     return false;
-  
+
```
Comment (dcci) [inline source/Plugins/Language/CPlusPlus/LibCxx.cpp:145]: Same here.

File: source/Plugins/Language/CPlusPlus/CPlusPlusLanguage.cpp (hunk 1 above)
Comment (shafik) [reply source/Plugins/Language/CPlusPlus/CPlusPlusLanguage.cpp:254]: Discussed and decided to leave it in place.

**Pull request merged**